code/modules/food_and_drinks/drinks/drinks/cans.dm

```diff
@@ -1,17 +1,45 @@
 /obj/item/reagent_containers/food/drinks/cans
-	var/canopened = 0
+	var/canopened = FALSE
 	var/is_glass = 0
 	var/is_plastic = 0
+	var/shaken = 0
+	var/bursting = FALSE
+	var/burstchance = 0
 
 /obj/item/reagent_containers/food/drinks/cans/New()
 	..()
 	flags &= ~OPENCONTAINER
 
-/obj/item/reagent_containers/food/drinks/cans/attack_self(mob/user)
-	if(canopened == 0)
+/obj/item/reagent_containers/food/drinks/cans/attack_self(mob/user, burstopen = FALSE)
+	if(!canopened && shaken)
+		if(!burstopen)
+			to_chat(user, "<span class='notice'>You open the drink with an audible pop!</span>")
+		else
+			visible_message("<span class='warning'>The [name] bursts open!</span>")
+		if(shaken < 5)
+			to_chat(user, "<span class='warning'>The [name] fizzes violently!</span>")
+		else
+			to_chat(user, "<span class='warning'>The [name] erupts into foam!</span>")
+			foam_up()
+
+		for(var/mob/living/carbon/C in range(1, get_turf(src)))
+			C.visible_message("<span class='warning'>You are splattered with [name]!</span>")
```

Comment:
This will show the message to everybody whenever somebody gets splattered.
When X gets splattered Y gets to see this message